src/tracker_manager.cpp

```diff
@@ -241,16 +241,17 @@ namespace libtorrent
 	void tracker_manager::remove_request(tracker_connection const* c)
 	{
 		TORRENT_ASSERT(is_single_thread());
-		http_conns_t::iterator i = std::find_if(m_http_conns.begin(), m_http_conns.end()
+		auto const i = std::find_if(m_http_conns.begin(), m_http_conns.end()
 			, [c] (std::shared_ptr<http_tracker_connection> const& ptr) { return ptr.get() == c; });
 		if (i != m_http_conns.end())
 		{
 			m_http_conns.erase(i);
 			return;
 		}
 
-		udp_conns_t::iterator j = std::find_if(m_udp_conns.begin(), m_udp_conns.end()
-			, [c] (udp_conns_t::value_type const& uc) { return uc.second.get() == c; });
+		auto const j = std::find_if(m_udp_conns.begin(), m_udp_conns.end()
+			, [c](std::pair<std::uint32_t, std::shared_ptr<udp_tracker_connection>> const& uc)
```

Comment:
at a higher level, it seems a bit suspicious to search an unordered_map like this. I would expect the ``tracker_connection`` object to know its key for an efficient lookup here.